tracing/src/macros.rs

```diff
@@ -783,6 +783,242 @@ macro_rules! event {
     );
 }
 
+/// Check's whether an equivalent `event!` invocation would result in an enabled
+/// event.
+///
+/// If you are using this macro to guard a log line that requires expensive computation, it can
+/// result in false-negative's, if the default collector has filters based on line numbers or field
+/// names.
+///
+/// This macro operates similarly to [`event!`], with some extensions:
+/// - Allows passing just a level.
+/// - Allows passing just a level and a target.
+///
+/// See [the top-level documentation][lib] for details on the syntax accepted by
+/// this macro.
+///
+/// [lib]: crate#using-the-macros
+/// [`event!`]: event!
+///
+/// # Examples
+///
+/// ```rust
+/// use tracing::{enabled, Level};
+///
+/// # fn main() {
+/// #   if enabled!(Level::DEBUG, "Debug loggin") {
+/// #       // Do something expensive
+/// #   }
+/// # }
+/// ```
+///
+#[macro_export]
+macro_rules! enabled {
+    (target: $target:expr, parent: $parent:expr, $lvl:expr, { $($fields:tt)* } )=> (
```

Comment:
I don't think this should take a parent span. As I discussed in my comment on the new `Event` constructors you added, the `Collect::enabled` method doesn't care about overridden span parents. The macro should not include a parent span.

Reply:
oh I see here now, sounds good!